src/AsyncProcess/index.ts

```diff
@@ -173,14 +220,44 @@ export class AsyncProcess<TIdentifier extends string> {
    * Return a boolean according to registered predicates values.
    */
   async shouldStart(): Promise<boolean> {
-    for (const predicateFn of this._predicateFns) {
-      if (!(await predicateFn(this))) {
-        return false
+    for (const [identifier, predicateFns] of this._fns.predicateFns.entries()) {
+      for (const predicateFn of predicateFns) {
+        if (!(await predicateFn(this))) {
+          this._log('functionsExecutions')('debug')(
+            `Skip jobs execution because of the "${identifier}" predicate`
+          )
+
+          return false
+        }
       }
     }
+
     return true
   }
 
+  /**
+   * Delete functions.
+   */
+  shouldDeleteFunctions(): this {
+    if (!this._options.deleteFunctionsWhenJobsStarted) {
```

Comment:
You called the option "deleteFunctionsWhenJobsStarted", but in facts, you always call the shouldDeleteFunctions after the call of each functions in the onSuccessFns map. 

Maybe this should be more accurate to name it "deleteFunctionsWhenJobsSucceeded", wdyt ?

Reply:
The idea is to avoid functions registered several times when "configuring" again the Async Process instance. So when you call the `start()` methods, whether if the success or the fail or even the skip (via a predicate) of the job(s), functions have been deleted. So you can register properly again the next time.